pep-0484.txt

```diff
@@ -276,6 +276,33 @@ implemented by deferring to ``isinstance(x, collections.abc.Callable)``.
 However, ``isinstance(x, typing.Callable[...])`` is not supported.
 
 
+Decorators
+----------
+
+Decorators can modify the types of the functions or classes they
+decorate. To annotate a decorated declaration with the eventual
+decorated type, put a type comment on the line with the decorator::
+
+  from typing import ContextManager, Generator
+  from contextlib import contextmanager
+
+  class DatabaseSession: ...
+
+  @contextmanager  # type: Callable[[str], ContextManager[DatabaseSession]]
```

Comment:
Cool, yes, good distinction between cast and declaration.

Of those possibilities, I like the `decorated_type` wording best. 

I definitely like the full `Callable`-included syntax, because 
* it doesn't introduce any new concepts 
* there's no requirement that the type of a decorated function be a function
* it applies equally well to decorated classes, if we need such a thing there.